src/File/RenameUpload.php

```diff
@@ -237,26 +107,42 @@ public function filter(mixed $value): mixed
         }
 
         // File upload via traditional SAPI
-        if (is_array($value) && isset($value['tmp_name'])) {
-            return $this->filterSapiUploadedFile($value);
+        if (
+            is_array($value)
+            && array_key_exists('tmp_name', $value)
+            && array_key_exists('name', $value)
+        ) {
+            /** @var UploadedFile $uploadedFile */
+            $uploadedFile = $value;
+            return $this->filterSapiUploadedFile($uploadedFile);
         }
 
         // String filename
         if (is_string($value)) {
             return $this->filterStringFilename($value);
         }
 
-        // Unrecognized; return verbatim
         return $value;
     }
 
     /**
-     * @param  string $sourceFile Source file path
-     * @param  string $targetFile Target file path
+     * @psalm-suppress MixedReturnStatement
+     * @template T
+     * @param T $value
+     * @return (T is UploadedFileInterface
```

Comment:
sadly i had to duplicate this doctype because otherwise psalm will detect the `mixed` return type and mark that as an error for all following calls in the Tests. This comes from the FilterInterface `@return TFilteredValue|T` but i could not define it strict. So i need to use `@implements FilterInterface<mixed>` in this class and overwrite the return type here manually